chuck-mqtt/src/main/java/com/gojek/chuckmqtt/internal/presentation/transactionlist/ui/screen/TransactionListScreen.kt

```diff
@@ -0,0 +1,171 @@
+        )
+    }
+}
+
+@Composable
+internal fun ListAppBar(
+    toolbarSubtitle: CharSequence,
+    intentLambda: (TransactionListIntent) -> Unit
+) {
+    TopAppBar(
+        title = {
+            Column {
+                Text("MQTT Chuck")
+                Text(
+                    text = toolbarSubtitle.toString(),
+                    style = TextStyle(fontSize = 16.sp, fontWeight = FontWeight.Normal)
+                )
+            }
+        },
+        actions = {
+            IconButton(onClick = { intentLambda(ClearTransactionHistoryIntent) }) {
+                Icon(
+                    painter = painterResource(id = R.drawable.mqtt_chuck_ic_delete_white_24dp),
+                    contentDescription = "Delete Packets"
+                )
+            }
+        },
+        backgroundColor = MaterialTheme.colors.primarySurface,
+        elevation = 4.dp
+    )
+}
+
+@Composable
+internal fun TransactionList(
+    transactionList: List<MqttTransactionUiModel>,
+    intentLambda: (TransactionListIntent) -> Unit
+) {
+    LazyColumn {
+        items(transactionList) { item ->
+            TransactionListItem(item = item, intentLambda = intentLambda)
+            Divider(color = Color.LightGray)
+        }
+    }
+}
+
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+internal fun TransactionListItem(
+    item: MqttTransactionUiModel,
+    intentLambda: (TransactionListIntent) -> Unit
+) {
+    val iconDrawable = if (item.isSent) {
+        R.drawable.mqtt_ic_message_sent
+    } else {
+        R.drawable.mqtt_ic_message_received
+    }
+
+    Surface(
+        onClick = {
+            intentLambda(OpenTransactionDetailIntent(item.id))
+        },
+        modifier = Modifier.fillMaxWidth()
+    ) {
+        Row(
+            modifier = Modifier.padding(8.dp)
+        ) {
+            Image(
+                modifier = Modifier.align(Alignment.CenterVertically),
+                painter = painterResource(id = iconDrawable),
+                contentDescription = if (item.isSent) {
+                    "Message Sent"
+                } else {
+                    "Message Received"
+                }
+            )
+
+            Spacer(modifier = Modifier.width(8.dp))
+
+            Column {
+                with(item) {
+                    Text(
+                        text = packetName,
+                        style = TextStyle(
+                            fontWeight = FontWeight.Bold,
+                            fontSize = 16.sp
+                        )
+                    )
+
+                    Text(
+                        text = packetInfo,
+                        color = Color(0xFF494A4A),
+                        style = TextStyle(
+                            fontWeight = FontWeight.Normal,
+                            fontSize = 14.sp
+                        )
+                    )
+
+                    Text(
+                        transmissionTime,
+                        color = Color(0xFF494A4A),
```

Comment:
Lets use colors through theme to support dark mode

Reply:
Can we define the colors first for both Light and Dark mode? I can see the Color Palette that you have defined here, but:
- Both `primaryVariant` and `secondary` colors are same for Light and Dark (while they should be different based on Light or Dark mode)
- Not enough contrast between Light's `primary` with Dark's

<img width="366" alt="Screen Shot 2022-06-14 at 13 05 36" src="https://user-images.githubusercontent.com/6652893/173504919-796fbb1e-03e4-4611-a7ab-718583baead5.png">

Reference: https://material.io/design/color/the-color-system.html#color-theme-creation
 
Once we have the colors, then we can make use of it on Composables.